src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
same as the previous function, we can remove the ternary implementation and use the `filter` method, using auxiliary variables to store the mother's data and let the code be more readable.

```js
const mothers = people.map((m) => m.name);

const children = people.filter((p) => {
  const hasMother = mothers.includes(p.mother);
  if (onlyWithSon) {
    return p.sex === 'm' && hasMother;
  }
  return hasMother;
});
```